### ISSUE TITLE
Get a bug when @extend in double @for

### ISSUE THREAD

**Issue by qiu8310** [sass/node-sass#1442]
```
$MAX_COL_NUMBER: 3;
$MAX_ROW_NUMBER: 3;

@for $i from 1 through $MAX_ROW_NUMBER {
  %order#{$i} {
    order: $i;
  }
}

@for $r from 1 through $MAX_ROW_NUMBER {
  @for $c from 1 through $MAX_COL_NUMBER {
    @if $r >= $c {
      .tableOf#{$r}Rows .cell {
        &:nth-child(#{$r + if($r == $c, "n", "n+" + $c)}) {
          // order: $r;
          @extend %order#{$r};
        }
      }
    }
  }
}
```

will output 

```
.tableOf1Rows .cell:nth-child(1n) {
  order: 1; }

.tableOf2Rows .cell:nth-child(2n+1) {
  order: 2; }

.tableOf3Rows .cell:nth-child(3n+1) {
  order: 3; }
```

but expect 

```
.tableOf1Rows .cell:nth-child(1n) {
  order: 1; }

.tableOf2Rows .cell:nth-child(2n+1) {
  order: 2; }

.tableOf2Rows .cell:nth-child(2n) {
  order: 2; }

.tableOf3Rows .cell:nth-child(3n+1) {
  order: 3; }

.tableOf3Rows .cell:nth-child(3n+2) {
  order: 3; }

.tableOf3Rows .cell:nth-child(3n) {
  order: 3; }
```

**Comment by xzyfer** [comment]
Does this still happen with the beta?

npm install node-sass@beta

**Comment by qiu8310** [comment]
It not work neither
![image](https://cloud.githubusercontent.com/assets/1094697/14414911/f1ffe7b4-ffcb-11e5-914b-9d8646c8a460.png)

**Comment by xzyfer** [comment]
This is fixed on master and will be in 3.5.0.

**Comment by patrickberkeley** [comment]
@xzyfer Has this shipped? We have node-sass 3.5.2 install on heroku and are seeing this issue.

**Comment by xzyfer** [comment]
@patrickberkeley the specific example in this issue has been fixed. If you're still experiencing issue please open a bug with [LibSass](https://github.com/sass/libsass) **only** after confirming the bug isn't in your code by compiling it with Ruby Sass. You can do this online at sassmeister.com.
